tensorflow_addons/image/tests/color_ops_test.py

```diff
@@ -50,6 +51,22 @@ def test_equalize_with_PIL():
     np.testing.assert_equal(color_ops.equalize(tf.constant(image)).numpy(), equalized)
 
 
+@pytest.mark.usefixtures("maybe_run_functions_eagerly")
+def test_equalize_with_skimage():
+    np.random.seed(0)
+    image = np.random.randint(low=0, high=256, size=(5, 5, 3, 3), dtype=np.uint8)
+    bins = np.random.randint(low=256, dtype=np.int32)
+    equalized = tf.cast(color_ops.equalize(tf.constant(image), bins), np.float16)
+    equalized = tf.math.divide(equalized, 255)
+    skiequalized = equalize_hist(image, bins)
+    assert (
+        tf.reduce_mean(
+            tf.math.abs(tf.reshape(tf.math.subtract(equalized, skiequalized), [-1]))
+        )
+        <= 0.015
+    ), "Greater than 1.5% difference in arrays"
```

Comment:
I didn't go into details but I'm still a little bit concerned about these numerical differences on a such small input matrix. @WindQAQ What do you think?

Reply:
The 1.5% metric comes from the difference between the implementations of the proven `PIL.ImageOps.equalize` and the test I'm using with `skimage.exposure.equalize_hist`.

Here's a quick showcase I made
```
#compares PIL and skimage
import tensorflow as tf
import numpy as np
import PIL
from skimage.exposure import equalize_hist
from PIL.ImageOps import *

np.random.seed(0)

low_image = np.random.randint(low=0, high=256, size=(4, 3, 3, 3), dtype=np.uint8)
PIL_equalized = np.stack([PIL.ImageOps.equalize(PIL.Image.fromarray(i)) for i in low_image])
PIL_equalized = tf.math.divide(tf.cast(PIL_equalized, tf.float32), 255) #normalized
skiequalized = equalize_hist(low_image, 256)
low_avg = tf.reduce_mean(
            tf.math.abs(tf.reshape(tf.math.subtract(PIL_equalized, skiequalized), [-1]))
)

high_image = np.random.randint(low=0, high=256, size=(28, 28, 3, 3), dtype=np.uint8)
PIL_equalized = np.stack([PIL.ImageOps.equalize(PIL.Image.fromarray(i)) for i in high_image])
PIL_equalized = tf.math.divide(tf.cast(PIL_equalized, tf.float32), 255) #normalized
skiequalized = equalize_hist(high_image, 256)
high_avg = tf.reduce_mean(
            tf.math.abs(tf.reshape(tf.math.subtract(PIL_equalized, skiequalized), [-1]))
)
print(low_avg, high_avg)
```
```
OUTPUT:
tf.Tensor(0.036022358, shape=(), dtype=float32) tf.Tensor(0.0061728833, shape=(), dtype=float32)
```
Here the differences between the two ranges from 3.6-0.6% depending on the amount of variance allowed by size of the image/number of pixels compared. So 1.5% is a reasonable metric that stands between this range that accounts for the different implementations

Reply:
Please add a comment in the test why we need this tolerance

Reply:
comment has been added